crates/aranya-client/src/aqc/api.rs

```diff
@@ -45,38 +56,48 @@ impl<'a> AqcChannels<'a> {
     /// # Cancellation Safety
     ///
     /// This method is NOT cancellation safe. Cancelling the resulting future might lose data.
-    #[instrument(skip_all, fields(%team_id, %peer, %label_id))]
-    pub async fn create_bidi_channel(
+    #[instrument(skip_all, fields(%team_id, %label_id))]
+    pub async fn create_bidi_channel<I>(
         &mut self,
         team_id: TeamId,
-        peer: NetIdentifier,
+        peer: I,
         label_id: LabelId,
-    ) -> crate::Result<AqcBidiChannel> {
+    ) -> crate::Result<AqcBidiChannel>
+    where
+        I: TryInto<NetIdentifier>,
+    {
         debug!("creating bidi channel");
 
+        let peer: NetIdentifier = peer.try_into().map_err(|_| InvalidNetIdentifier(()))?;
+
         let (aqc_ctrl, psks) = self
             .client
             .daemon
-            .create_aqc_bidi_channel(context::current(), team_id, peer.clone(), label_id)
+            .create_aqc_bidi_channel(
+                context::current(),
+                team_id.into_api(),
+                peer.clone().into_api(),
+                label_id.into_api(),
+            )
             .await
             .map_err(IpcError::new)?
             .map_err(aranya_error)?;
         debug!(%label_id, num_psks = psks.len(), "created bidi channel");
 
-        let peer_addr = tokio::net::lookup_host(peer.0.as_str())
+        let peer_addr = tokio::net::lookup_host(peer.as_ref())
             .await
             .map_err(AqcError::AddrResolution)?
             .next()
             .ok_or_else(no_addr)?;
 
         self.client
             .aqc
-            .send_ctrl(peer_addr, aqc_ctrl, team_id)
+            .send_ctrl(peer_addr, aqc_ctrl, team_id.into_api())
```

Comment:
Would be convenient if we could simplify this to:
```suggestion
            .send_ctrl(peer_addr, aqc_ctrl, team_id.into())
```